include/dma.h

```diff
@@ -151,7 +151,7 @@ struct dma_config {
 	struct dma_block_config *head_block;
 	void *callback_arg;
 	void (*dma_callback)(void *callback_arg, u32_t channel,
-			     int error_code);
+			     int ret_code);
```

Comment:
This is not compliant with current DMA API:
```
 * dma_callback is the callback function pointer. If enabled, callback function
 *              will be invoked at transfer completion or when error happens
 *              (error_code: zero-transfer success, non zero-error happens).
```
I'm not opposed to the change, but it would require dma API alignment

Reply:
Ho, I missed that  sorry, should not start review before coffee.

Reply:
This means that all code using the STM32F4 DMA (I2S driver for example) must be changed as well.
Shouldn't be part of the PR as well?

Reply:
I noticed that you change the dma_callback(0 signature here, but not in "struct dma_stm32_stream" in c file.
I guess you should do that as well.